sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -993,14 +993,34 @@ colPosition
     : position=FIRST | position=AFTER afterCol=errorCapturingIdentifier
     ;
 
+type
+    : BOOLEAN
+    | TINYINT | BYTE
+    | SMALLINT | SHORT
+    | INT | INTEGER
+    | BIGINT | LONG
+    | FLOAT | REAL
+    | DOUBLE
+    | DATE
+    | TIMESTAMP | TIMESTAMP_NTZ | TIMESTAMP_LTZ
+    | STRING
+    | CHARACTER | CHAR
+    | VARCHAR
+    | BINARY
+    | DECIMAL | DEC | NUMERIC
+    | VOID
+    | INTERVAL
+    | ARRAY | STRUCT | MAP
+    ;
```

Comment:
shall we still allow identifier? then we can keep the previous nice error message for supported type string.

Reply:
`identifier` doesn't catch any bad input as it is defined:
```
IDENTIFIER
    : (LETTER | DIGIT | '_')+
    ;
```

Even on Spark 3.3, sometimes we get `Syntax error`:
```sql
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 3.3.2
      /_/

scala> sql("CREATE TABLE tbl(a #abc)")
org.apache.spark.sql.catalyst.parser.ParseException:
Syntax error at or near '#': extra input '#'(line 1, pos 19)

== SQL ==
CREATE TABLE tbl(a #abc)
-------------------^^^
```
or sometimes:
```sql
scala> sql("CREATE TABLE tbl(a 1abc)")
org.apache.spark.sql.catalyst.parser.ParseException:
DataType 1abc is not supported.(line 1, pos 19)
```

Reply:
The previous code uses `identifier`, if we keep it, we can at least  be on-par with the previous error message.

Reply:
ok. done.